### PR TITLE
Implements remote storage and FieldFile support

### PR DESCRIPTION
**Pull request by MRigal**

### CODE REVIEW

File: sendfile/__init__.py [hunk 1]
```diff
@@ -53,21 +80,32 @@ def sendfile(request, filename, attachment=False, attachment_filename=None, mime
     filename (using the standard python mimetypes module)
     '''
     _sendfile = _get_sendfile()
+    filename = normalize_filename(filename)
+    remote_storage = is_remote_storage(filename)
 
-    if not os.path.exists(filename):
+    if not remote_storage and not os.path.exists(filename):
```
Comment (MRigal) [inline sendfile/__init__.py]: Refactor to use one big "if not remote_storage" to improve readability

File: sendfile/__init__.py [hunk 2]
```diff
@@ -53,21 +80,32 @@ def sendfile(request, filename, attachment=False, attachment_filename=None, mime
...
-    if mimetype is None:
-        if guessed_mimetype:
-            mimetype = guessed_mimetype
-        else:
-            mimetype = 'application/octet-stream'
-        
-    response = _sendfile(request, filename, mimetype=mimetype)
+    #else:
+    #    remote_storage = True
+    # TODO: check for url and set remote_storage to True
```
Comment (MRigal) [inline sendfile/__init__.py]: cleanup

File: sendfile/__init__.py [hunk 3]
```diff
@@ -29,12 +36,32 @@ def _get_sendfile():
...
+
     if not backend:
         raise ImproperlyConfigured('You must specify a value for SENDFILE_BACKEND')
     module = import_module(backend)
     return module.sendfile
 
 
+def normalize_filename(filename):
+    """
+    Fielename is always a string (path or url)
```
Comment (MRigal) [inline sendfile/__init__.py]: typo filename

File: sendfile/__init__.py [hunk 4]
```diff
@@ -53,21 +80,32 @@ def sendfile(request, filename, attachment=False, attachment_filename=None, mime
...
+        if mimetype is None:
+            if guessed_mimetype:
+                mimetype = guessed_mimetype
+            else:
+                mimetype = 'application/octet-stream'
+
+    response = _sendfile(request, filename, remote_storage=remote_storage, mimetype=mimetype)
     if attachment:
         if attachment_filename is None:
+            # This will also work with URLs from a remote storage, so will be left untouched.
```
Comment (MRigal) [inline sendfile/__init__.py]: Rewrite comment to Remove will and future as this is now state of the art

File: sendfile/__init__.py [hunk 5]
```diff
@@ -77,19 +115,21 @@ def sendfile(request, filename, attachment=False, attachment_filename=None, mime
...
         response['Content-Disposition'] = '; '.join(parts)
 
-    response['Content-length'] = os.path.getsize(filename)
-    response['Content-Type'] = mimetype
-    if not encoding:
-        encoding = guessed_encoding
-    if encoding:
-        response['Content-Encoding'] = encoding
+    if not remote_storage:
+
```
Comment (MRigal) [inline sendfile/__init__.py]: remove emptyline

File: sendfile/backends/simple.py [hunk 6]
```diff
@@ -7,22 +7,28 @@
...
 def sendfile(request, filename, **kwargs):
-    # Respect the If-Modified-Since header.
-    statobj = os.stat(filename)
+    remote_storage = kwargs.get('remote_storage', False)
 
-    if not was_modified_since(request.META.get('HTTP_IF_MODIFIED_SINCE'),
-                              statobj[stat.ST_MTIME], statobj[stat.ST_SIZE]):
-        return HttpResponseNotModified()
+    if remote_storage:
+        response = HttpResponseRedirect(filename)
```
Comment (MRigal) [inline sendfile/backends/simple.py]: return here or also set next return to a response

**Comment by hckjck** [comment]
Merged. Thanks for review.

**Pull request closed without merging**